when the minio container is already running I get "The container name "/minio-minio" is already in use "

membrane error, exiting: ContainerCreate: Error response from daemon: Conflict. The container name "/minio-minio" is already in use by container "478dc1132433a1487ef14f6d6218bf85a26253c7a101a88e1927c05bf8496a0e". You have to remove (or rename) that container to be able to reuse that name.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0x2764d39]

goroutine 1 [running]:
github.com/nitrictech/nitric/pkg/membrane.(*Membrane).Stop(0x0)
	/home/runner/go/pkg/mod/github.com/nitrictech/nitric@v0.13.0-rc.11/pkg/membrane/membrane.go:278 +0x19
github.com/nitrictech/cli/pkg/run.(*localServices).Stop(0xc00038a500)
	/home/runner/work/cli/cli/pkg/run/run.go:71 +0x25
github.com/nitrictech/cli/pkg/cmd/run.glob..func1(0x5498800, {0xc0000ce3d0, 0x1, 0x1})
	/home/runner/work/cli/cli/pkg/cmd/run/root.go:157 +0xeb3
github.com/spf13/cobra.(*Command).execute(0x5498800, {0xc0000ce3b0, 0x1, 0x1})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:860 +0x5f8
github.com/spf13/cobra.(*Command).ExecuteC(0x5499700)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902
main.Execute()
	/home/runner/work/cli/cli/pkg/cmd/root.go:61 +0x32
main.main()
	/home/runner/work/cli/cli/pkg/cmd/main.go:20 +0x17

--- a/mocks/mock_containerengine/mock_containerengine.go
+++ b/mocks/mock_containerengine/mock_containerengine.go
@@ -199,17 +199,17 @@ func (mr *MockContainerEngineMockRecorder) NetworkCreate(arg0 interface{}) *gomo
 }
 
 // RemoveByLabel mocks base method.
-func (m *MockContainerEngine) RemoveByLabel(arg0, arg1 string) error {
+func (m *MockContainerEngine) RemoveByLabel(arg0 map[string]string) error {
 	m.ctrl.T.Helper()
-	ret := m.ctrl.Call(m, "RemoveByLabel", arg0, arg1)
+	ret := m.ctrl.Call(m, "RemoveByLabel", arg0)
 	ret0, _ := ret[0].(error)
 	return ret0
 }
 
 // RemoveByLabel indicates an expected call of RemoveByLabel.
-func (mr *MockContainerEngineMockRecorder) RemoveByLabel(arg0, arg1 interface{}) *gomock.Call {
+func (mr *MockContainerEngineMockRecorder) RemoveByLabel(arg0 interface{}) *gomock.Call {
 	mr.mock.ctrl.T.Helper()
-	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "RemoveByLabel", reflect.TypeOf((*MockContainerEngine)(nil).RemoveByLabel), arg0, arg1)
+	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "RemoveByLabel", reflect.TypeOf((*MockContainerEngine)(nil).RemoveByLabel), arg0)
 }
 
 // Start mocks base method.

--- a/pkg/containerengine/docker.go
+++ b/pkg/containerengine/docker.go
@@ -276,12 +276,14 @@ func (d *docker) ContainersListByLabel(match map[string]string) ([]types.Contain
 	return d.cli.ContainerList(context.Background(), opts)
 }
 
-func (d *docker) RemoveByLabel(name, value string) error {
+func (d *docker) RemoveByLabel(labels map[string]string) error {
 	opts := types.ContainerListOptions{
 		All:     true,
 		Filters: filters.NewArgs(),
 	}
-	opts.Filters.Add("label", fmt.Sprintf("%s=%s", name, value))
+	for name, value := range labels {
+		opts.Filters.Add("label", fmt.Sprintf("%s=%s", name, value))
+	}
 
 	res, err := d.cli.ContainerList(context.Background(), opts)
 	if err != nil {

--- a/pkg/containerengine/podman.go
+++ b/pkg/containerengine/podman.go
@@ -121,8 +121,8 @@ func (p *podman) ContainersListByLabel(match map[string]string) ([]types.Contain
 	return p.docker.ContainersListByLabel(match)
 }
 
-func (p *podman) RemoveByLabel(name, value string) error {
-	return p.docker.RemoveByLabel(name, value)
+func (p *podman) RemoveByLabel(labels map[string]string) error {
+	return p.docker.RemoveByLabel(labels)
 }
 
 func (p *podman) ContainerExec(containerName string, cmd []string, workingDir string) error {

--- a/pkg/containerengine/types.go
+++ b/pkg/containerengine/types.go
@@ -55,7 +55,7 @@ type ContainerEngine interface {
 	ContainerWait(containerID string, condition container.WaitCondition) (<-chan container.ContainerWaitOKBody, <-chan error)
 	CopyFromArchive(nameOrID string, path string, reader io.Reader) error
 	ContainersListByLabel(match map[string]string) ([]types.Container, error)
-	RemoveByLabel(name, value string) error
+	RemoveByLabel(labels map[string]string) error
 	ContainerExec(containerName string, cmd []string, workingDir string) error
 	ContainerLogs(containerID string, opts types.ContainerLogsOptions) (io.ReadCloser, error)
 	Logger(stackPath string) ContainerLogger

--- a/pkg/run/minio.go
+++ b/pkg/run/minio.go
@@ -44,12 +44,10 @@ const (
 	minioImage       = "minio/minio:latest"
 	devVolume        = "/nitric/"
 	runPerm          = os.ModePerm // NOTE: octal notation is important here!!!
-	LabelRunID       = "io.nitric-run-id"
-	LabelStackName   = "io.nitric-stack"
-	LabelType        = "io.nitric-type"
+	labelStackName   = "io.nitric/stack"
+	labelType        = "io.nitric/type"
 	minioPort        = 9000
 	minioConsolePort = 9001 // TODO: Determine if we would like to expose the console
-
 )
 
 // StartMinio -
@@ -88,6 +86,10 @@ func (m *MinioServer) Start() error {
 			nat.Port(fmt.Sprintf("%d/tcp", minioPort)):        struct{}{},
 			nat.Port(fmt.Sprintf("%d/tcp", minioConsolePort)): struct{}{},
 		},
+		Labels: map[string]string{
+			labelStackName: m.name,
+			labelType:      "minio",
+		},
 	}
 
 	hc := &container.HostConfig{
@@ -115,7 +117,9 @@ func (m *MinioServer) Start() error {
 		NetworkMode: container.NetworkMode("bridge"),
 	}
 
-	cID, err := m.ce.ContainerCreate(cc, hc, &network.NetworkingConfig{EndpointsConfig: map[string]*network.EndpointSettings{}}, "minio-"+m.name)
+	cID, err := m.ce.ContainerCreate(cc, hc, &network.NetworkingConfig{
+		EndpointsConfig: map[string]*network.EndpointSettings{},
+	}, "minio-"+m.name)
 	if err != nil {
 		return err
 	}
@@ -140,6 +144,15 @@ func NewMinio(dir string, name string) (*MinioServer, error) {
 	ce, err := containerengine.Discover()
 	if err != nil {
 		return nil, err
+	}
+
+	// Remove any existing containers with this label.
+	err = ce.RemoveByLabel(map[string]string{
+		labelStackName: name,
+		labelType:      "minio",
+	})
+	if err != nil {
+		return nil, errors.WithMessage(err, "could not remove existing minio container")
 	}
 
 	return &MinioServer{

--- a/pkg/run/run.go
+++ b/pkg/run/run.go
@@ -87,7 +87,7 @@ func (l *localServices) Status() *LocalServicesStatus {
 
 func (l *localServices) Start(pool worker.WorkerPool) error {
 	var err error
-	l.mio, err = NewMinio(l.status.RunDir, "minio")
+	l.mio, err = NewMinio(l.status.RunDir, l.stackName)
 	if err != nil {
 		return err
 	}

